pyteal/__init__.pyi

```diff
@@ -45,6 +45,13 @@ __all__ = [
     "BitwiseOr",
     "BitwiseXor",
     "Block",
+    "BoxCreate",
```

Comment:
@ahangsu The functionality looks good to me though I think the user guide ought to reflect boxes in https://pyteal.readthedocs.io/en/stable/state.html.  What's your preference?  Merge as is + open new PR focused only on user guide or include user guide here.

* Either works for me - Looking to gauge the review.
* If helpful to discuss user guide content, I'm happy to talk live - let me know.

Reply:
I would like to finish the last piece of docs in this PR, to let my implementation sit with the document. We should talk about it at some time, that will help me to phrase the documentation better.